src/cpp/core/Error.cpp

```diff
@@ -155,6 +156,20 @@ bool Error::isError() const
       return false ;
 }
 
+bool Error::isShutdownError(const boost::system::error_code& ec)
```

Comment:
For better or worse SocketUtils.hpp has it's functions (which are very similar to this one) defined in the http namespace so let's stick with that for the sake of consistency.